src/core/include/units/quantity.h

```diff
@@ -51,6 +51,71 @@ concept quantity_one =
   (std::same_as<typename T::dimension, dim_one> || std::same_as<typename T::dimension, unknown_dimension<>>) &&
   detail::equivalent_unit<typename T::unit, typename T::dimension, ::units::one, typename T::dimension>::value;
 
+template<typename T>
+concept has_noexcept_math = requires(T t) {
```

Comment:
This is overconstraining. `+` may be `noexcept` while `*` may throw for some `Rep`.

Reply:
Besides, this makes operators conditionally `noexcept` which is inconsistent with the guidelines.